bin/version-packages.sh

```diff
@@ -50,13 +51,19 @@ fi
 # Bail if no composer.json to check for.
 if [[ ! -f "$CURRENT_DIR/composer.json" ]]; then
     echo "EXITING: This script must be run from a directory with composer.json at it's root."
-    exit;
+    exit 1;
+fi
+
+if [[ $COMPOSER_VER == "1."* ]]; then
+	echo "EXITING: This script requires Composer 2.0.0+"
+	exit 1;
 fi
 
 # Get the list of package names to update.
 # Works in accordance of `composer show`, and will only act on packages prefixed with `automattic/jetpack-`.
 # Using --self because it is agnostic to whether /vendor is populated.
-composer show --self |
+# Using --no-dev since we don't care about ensuring version of development packages.
+composer show --self --no-dev |
```

Comment:
This doesn't seem to work. `composer show --self --no-dev` still prints the require-dev dependencies.

It looks like `--no-dev` affects the list-style output, e.g. from `composer show` or `composer show automattic/jetpack-*`. It also prevents `composer show <package>` from finding the named package to show it if it was installed via require-dev.

Reply:
78baf0f4d and 1d230c8d2